paper-api/src/main/java/org/bukkit/World.java

```diff
@@ -1891,7 +1876,24 @@ default Iterable<? extends net.kyori.adventure.audience.Audience> audiences() {
      * @return the ray trace hit result, or <code>null</code> if there is no hit
      */
     @Nullable RayTraceResult rayTraceBlocks(io.papermc.paper.math.@NotNull Position start, @NotNull Vector direction, double maxDistance, @NotNull FluidCollisionMode fluidCollisionMode, boolean ignorePassableBlocks, @Nullable Predicate<? super Block> canCollide);
-    // Paper end
+
+    /**
+     * Performs a ray trace that checks for block collisions using the blocks'
+     * precise shape.
+     * <p>
+     * This may cause loading of chunks! Some implementations may impose
+     * artificial restrictions on the maximum distance.
+     *
+     * @param start the start position
+     * @param direction the ray direction
+     * @param maxDistance the maximum distance
+     * @param fluidCollisionMode the fluid collision mode
+     * @param blockCollisionMode the block collision mode
+     * @param canCollide predicate for blocks the ray can potentially collide
+     *     with, or <code>null</code> to consider all blocks
+     * @return the ray trace hit result, or <code>null</code> if there is no hit
+     */
+    @Nullable RayTraceResult rayTraceBlocks(io.papermc.paper.math.@NotNull Position start, @NotNull Vector direction, double maxDistance, @NotNull FluidCollisionMode fluidCollisionMode, @NotNull io.papermc.paper.raytrace.BlockCollisionMode blockCollisionMode, @Nullable Predicate<? super Block> canCollide);
```

Comment:
The entire point of the builder was that we do not overload this further.
It is already *pain* xD

Reply:
The overloads are necessary for the builder to function?

Reply:
No, CraftWorld executes the builder.
It can/should use non public methods for that.
Overload the methods needed one last time in CraftWorld.
Any additions after that will just use the method you are about to add and add params.

Reply:
Alright, removed the method from `World`.